src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -92,7 +95,25 @@ def write(self, project: Project, rules: RuleSet) -> None:
     def merge(self, project: Project) -> None:
         """Read rules from all stores, merge and write them back so they all are up-to-date."""
         merged_rules = self.read(project)
-        self.write(project, merged_rules)
+        trimmed_rules = self._trim_rules(merged_rules)
+        self.write(project, trimmed_rules)
+
+    def _trim_rules(self, rules: RuleSet) -> RuleSet:
+        if self.MERGE_MAX_RULES < 0:
+            return rules
```

Comment:
nit: Since `MERGE_MAX_RULES` is a constant, this is dead code. That is, setting it to `-1` would require a code change, so we might as well remove this check.